CMakeLists.txt

```diff
@@ -205,44 +216,45 @@ else()
     PRIVATE  "$<BUILD_INTERFACE:${RAPIDS_TRITON_BACKEND_SOURCE_DIR}/include>"
              "${CMAKE_CURRENT_SOURCE_DIR}/src"
   )
-  
-  if(TRITON_ENABLE_GPU)
-    target_link_libraries(${BACKEND_TARGET}
-    PRIVATE
-      treelite::treelite_static
-      treelite::treelite_runtime_static
-      raft::raft
-      cuml++
-      rapids_triton::rapids_triton
-      triton-core-serverstub
-      triton-backend-utils
-      "${TRITONSERVER_LIB}"
-      $<TARGET_NAME_IF_EXISTS:conda_env>
-    )
-  else()
-    target_link_libraries(${BACKEND_TARGET}
-    PRIVATE
-      treelite::treelite_static
-      treelite::treelite_runtime_static
-      rapids_triton::rapids_triton
-      triton-core-serverstub
-      triton-backend-utils
-      "${TRITONSERVER_LIB}"
-      $<TARGET_NAME_IF_EXISTS:conda_env>
+
+  if (TRITON_FIL_ENABLE_TREESHAP)
+    target_compile_definitions(${BACKEND_TARGET}
+      PRIVATE
+        TRITON_FIL_ENABLE_TREESHAP=1
     )
   endif()
   
-  if(TRITON_ENABLE_GPU)
-    install(
-      TARGETS ${BACKEND_TARGET} cuml++
-      LIBRARY DESTINATION ${BACKEND_FOLDER}/${BACKEND_NAME}
-    )
+  if(NOT TRITON_FIL_USE_TREELITE_STATIC)
+    list(APPEND TREELITE_LIBS treelite::treelite treelite::treelite_runtime)
+    list(APPEND TREELITE_LIBS_NO_PREFIX treelite treelite_runtime)
   else()
-    install(
-      TARGETS ${BACKEND_TARGET}
-      LIBRARY DESTINATION ${BACKEND_FOLDER}/${BACKEND_NAME}
-    )
+    list(APPEND TREELITE_LIBS treelite::treelite_static treelite::treelite_runtime_static)
+    list(APPEND TREELITE_LIBS_NO_PREFIX treelite_static treelite_runtime_static)
   endif()
+
+  target_link_libraries(${BACKEND_TARGET}
+  PRIVATE
+    $<$<BOOL:${TRITON_ENABLE_GPU}>:cuml++>
+    ${TREELITE_LIBS}
+    rapids_triton::rapids_triton
+    triton-core-serverstub
+    triton-backend-utils
+    "${TRITONSERVER_LIB}"
```

Comment:
Unrelated to changes in the PR so we can address it later, but are this quotes necessary?

Reply:
Not quite sure, I believe it's an environment variable and not a cmake variable. Possibly the reason? cc @wphicks

Reply:
I honestly don't remember. If I have a moment later, I'll review and see if I can understand what I was thinking.